src/tools/clippy/tests/ui/transmute_ptr_to_ptr.rs

```diff
@@ -35,16 +35,16 @@ fn transmute_ptr_to_ptr() {
         // ref-ref transmutes; bad
         let _: &f32 = std::mem::transmute(&1u32);
         //~^ ERROR: transmute from a reference to a reference
-        let _: &f64 = std::mem::transmute(&1f32);
+        let _: &f32 = std::mem::transmute(&1f64);
         //~^ ERROR: transmute from a reference to a reference
         //:^ this test is here because both f32 and f64 are the same TypeVariant, but they are not
         // the same type
         let _: &mut f32 = std::mem::transmute(&mut 1u32);
         //~^ ERROR: transmute from a reference to a reference
         let _: &GenericParam<f32> = std::mem::transmute(&GenericParam { t: 1u32 });
         //~^ ERROR: transmute from a reference to a reference
-        let u8_ref: &u8 = &0u8;
-        let u64_ref: &u64 = unsafe { std::mem::transmute(u8_ref) };
+        let u64_ref: &u64 = &0u64;
+        let u8_ref: &u8 = unsafe { std::mem::transmute(u64_ref) };
```

Comment:
What's the reason for swapping the types in the Clippy tests? Doesn't this now test something else?

Reply:
> What's the reason for swapping the types in the Clippy tests?

This was going from `&u8` to `&u64` which is UB, this is now going from `&u64` to `&u8`.

> Doesn't this now test something else?

I don't think so. The first case was testing two different floats, it's still the case. As for the second case, I think it was the same but with different integers.

Reply:
Ah I see. I should have taken a closer look on what the PR is doing, instead of just looking at the Clippy changes. Thanks for the explanation!